kubectl-volsync/cmd/migration_create.go

```diff
@@ -72,34 +104,329 @@ func validateMigrationCreate(cmd *cobra.Command, args []string) error {
 	return nil
 }
 
-func doMigrationCreate(cmd *cobra.Command, args []string) error {
-	// Create the empty relationship
-	configDir, err := cmd.Flags().GetString("config-dir")
+func newMigrationCreate(cmd *cobra.Command) (*migrationCreate, error) {
+	mc := &migrationCreate{}
+	// build struct migrationRelationship from cmd line args
+	mr, err := newmigrationRelationship(cmd)
+	if err != nil {
+		return nil, err
+	}
+	mc.mr = mr
+	// build struct migrationRelationshipDestination from cmd line args
+	err = newMigrationRelationshipDestination(mc, cmd)
+	if err != nil {
+		return nil, err
+	}
+
+	return mc, nil
+}
+
+//nolint:funlen
+func newMigrationRelationshipDestination(mc *migrationCreate, cmd *cobra.Command) error {
+	mrd := &migrationRelationshipDestination{}
+	mc.mr.data.Destination = mrd
+
+	cm, err := cmd.Flags().GetString("copymethod")
+	if err != nil {
+		return fmt.Errorf("failed to fetch the copy method, err = %w", err)
+	}
+	mrd.Destination.CopyMethod = volsyncv1alpha1.CopyMethodType(cm)
+	if mrd.Destination.CopyMethod != volsyncv1alpha1.CopyMethodDirect &&
+		mrd.Destination.CopyMethod != volsyncv1alpha1.CopyMethodClone &&
+		mrd.Destination.CopyMethod != volsyncv1alpha1.CopyMethodSnapshot {
+		return fmt.Errorf("unsupported copymethod: %v", mrd.Destination.CopyMethod)
+	}
+
+	pvcname, err := cmd.Flags().GetString("pvcname")
+	if err != nil {
+		return fmt.Errorf("failed to fetch the pvcname, err = %w", err)
+	}
+	xcr, err := ParseXClusterName(pvcname)
+	if err != nil {
+		return fmt.Errorf("failed to parse cluster name from pvcname, err = %w", err)
+	}
+	mrd.PVCName = xcr.Name
+	mrd.Namespace = xcr.Namespace
+
+	mc.clientObject, err = newClient(mrd.Cluster)
```

Comment:
Creating the client in the middle of things makes it hard to test. For example, the test framework starts an API server, but to access it, you'll have to use the provided client.Client, not one that is constructed from the provided cluster context name.

I'd suggest refactoring into high level steps of:
- parse CLI
- get client
- validate CLI (args relative to each other and the current cluster state)
- update cluster state

These can be in the Run() function and may not get unit tested, but that should be ok w/ proper structure since it's basically straight line code.
The steps can each be unit tested individually.